includes/class-validate-by-domain-options.php

```diff
@@ -5,167 +5,166 @@
-		[ $this, 'whitelist_render' ],
-		'validate_by_domain',
-		'options_section'
-	);
-
-}
+	/**
+	 * Register the plugin settings, create fields
+	 */
+	function plugin_settings_init() {
+
+		$args = array(
+			'sanitize_callback' => [ $this, 'sanitize_input' ],
+		);
+
+		register_setting( 'validate_by_domain', 'validate_by_domain_settings', $args );
+
+		add_settings_section(
+			'options_section',
+			__( '', 'WordPress' ),
+			'',
+			'validate_by_domain'
+		);
+
+		add_settings_field(
+			'validate_enable',
+			__( 'Enable', 'WordPress' ),
+			[ $this, 'enable_render' ],
+			'validate_by_domain',
+			'options_section'
+		);
+
+		add_settings_field(
+			'validate_role',
+			__( 'Role', 'WordPress' ),
+			[ $this, 'role_render' ],
+			'validate_by_domain',
+			'options_section'
+		);
+
+		add_settings_field(
+			'validate_whitelist',
+			__( 'Whitelist', 'WordPress' ),
+			[ $this, 'whitelist_render' ],
+			'validate_by_domain',
+			'options_section'
+		);
+
+	}
 
 // todo: Check for the existence of the key before using it in a condition, in the render functions below (avoid Illegal string offset)
 
-/**
- * Render the options page enable field
- */
-function enable_render() {
+	/**
+	 * Render the options page enable field
+	 */
+	function enable_render() {
 
-// $options = get_option( 'validate_by_domain_settings' );
-?><!-- <input
+		$options = get_option( 'validate_by_domain_settings' );
+		?><input
         type='checkbox'
         name='validate_by_domain_settings[validate_enable]' <?php checked( $options['validate_enable'], 1 ); ?>
-        value='1'>--><?php
-}
+        value='1'><?php
+	}
 
-/**
- * Render the options page role selection field
- */
-function role_render() {
+	/**
+	 * Render the options page role selection field
+	 */
+	function role_render() {
+		$options = get_option( 'validate_by_domain_settings' );
+		( $options['validate_role'] ) ? $role = $options['validate_role'] : $role = 'Subscriber';
+		?> <select name='validate_by_domain_settings[validate_role]>'
+		<?php
+		wp_dropdown_roles( $role );
+		?></select> <?php
+	}
 
-	// $options = get_option( 'validate_by_domain_settings' );
-	?>
-    <!-- <select name='validate_by_domain_settings[validate_role]'>
-            <option value='1' <?php selected( $options['validate_role'], 1 ); ?>>Contributor</option>
-            <option value='2' <?php selected( $options['validate_role'], 2 ); ?>>Editor</option>
-            <option value='3' <?php selected( $options['validate_role'], 3 ); ?>>Author</option>
-            <option value='4' <?php selected( $options['validate_role'], 4 ); ?>>Subscriber</option>
-        </select> -->
+	/**
+	 * Render the options page whitelist field
+	 */
+	function whitelist_render() {
 
-	<?php
+		$options = get_option( 'validate_by_domain_settings' );
+		?>
+        <textarea cols='60' rows='15'
+                  name='validate_by_domain_settings[validate_whitelist]'><?php echo $options['validate_whitelist']; ?></textarea>
+		<?php
```

Comment:
the use of single enclosing quotes will be employed `$html = '<a href="somelink">some link</a>';` unless there is some php to be rendered inside `$html = "<a href='somelink'>{$value}</a>"`